rtnetlink/src/ns.rs

```diff
@@ -0,0 +1,223 @@
+                    Some(none_fs),
+                    mount_flags,
+                    none_p4,
+                )
+                .is_err()
+                {
+                    exit(-1);
+                }
+
+                mount_flags = nix::mount::MsFlags::empty();
+                mount_flags.insert(nix::mount::MsFlags::MS_SHARED);
+                mount_flags.insert(nix::mount::MsFlags::MS_REC);
+                if nix::mount::mount(
+                    Some(Path::new("")),
+                    dir_path,
+                    Some(none_fs),
+                    mount_flags,
+                    none_p4,
+                )
+                .is_err()
+                {
+                    exit(-1);
+                }
+
+                let ns_path = Path::new(&netns_path);
+
+                // creating the netns file
+                let fd = match nix::fcntl::open(ns_path, open_flags, Mode::empty()) {
+                    Ok(raw_fd) => raw_fd,
+                    Err(_) => exit(-1),
+                };
+
+                if nix::unistd::close(fd).is_err() {
+                    let _ = nix::unistd::unlink(ns_path);
+                    exit(-1)
+                }
+
+                // unshare to the new network namespace
+                if nix::sched::unshare(CloneFlags::CLONE_NEWNET).is_err() {
+                    let _ = nix::unistd::unlink(ns_path);
+                    exit(-1);
+                }
+                open_flags = OFlag::empty();
+                open_flags.insert(OFlag::O_RDONLY);
+                open_flags.insert(OFlag::O_CLOEXEC);
+
+                let fd = match nix::fcntl::open(Path::new(&SELF_NS_PATH), open_flags, Mode::empty())
+                {
+                    Ok(raw_fd) => raw_fd,
+                    Err(_) => exit(-1),
+                };
+
+                let self_path = Path::new(&SELF_NS_PATH);
+
+                // bind to the netns
+                if nix::mount::mount(
+                    Some(self_path),
+                    ns_path,
+                    Some(none_fs),
+                    nix::mount::MsFlags::MS_BIND,
+                    none_p4,
+                )
+                .is_err()
+                {
+                    let _ = nix::unistd::unlink(ns_path);
+                    exit(-1);
+                }
+
+                setns_flags.insert(CloneFlags::CLONE_NEWNET);
+                if nix::sched::setns(fd, setns_flags).is_err() {
+                    let _ = nix::unistd::unlink(ns_path);
+                    exit(-1);
+                }
+
+                exit(0)
+            }
+            Err(e) => {
+                let err_msg = format!("Fork failed: {}", e);
+                Err(Error::NamespaceError(err_msg))
+            }
+        }
+    }
+
+    /// Remove a network namespace
+    /// This is equivalent to `ip netns del NS_NAME`.
+    pub async fn del(ns_name: String) -> Result<(), Error> {
+        let jh = task::spawn_blocking(move || {
```

Comment:
What does `jh` stand for here?

Reply:
It stands for job handler, I should have named it result or res. 
I'll rename it